pgx_lwt/src/s.ml

```diff
@@ -0,0 +1,21 @@
+module type IO = sig
+  type in_channel
+  type out_channel
+
+  type sockaddr =
+    | Unix of string
+    | Inet of string * int
+
+  val output_char : out_channel -> char -> unit Lwt.t
+  val output_string : out_channel -> string -> unit Lwt.t
+  val flush : out_channel -> unit Lwt.t
+  val input_char : in_channel -> char Lwt.t
+  val really_input : in_channel -> bytes -> int -> int -> unit Lwt.t
+  val close_in : in_channel -> unit Lwt.t
+  val getlogin : unit -> string Lwt.t
+  val open_connection : sockaddr -> (in_channel * out_channel) Lwt.t
+end
+
+module type Pgx_impl = sig
+  include Pgx.S with type 'a IO.t = 'a Lwt.t
+end
```

Comment:
Why do we need this? Isn't this just `Pgx.S with type IO.t := Lwt.t`?

Reply:
Good catch. I had initially done this as i need to return a first class module from the pgx_lwt_mirage.connect function and parameterized types aren't supported for packed modules. But I didn't realize that I can just create a temporary alias inside the mirage backend an then i won't need to create this temporary instance. I'll update the PR

Reply:
The module alias is now just Pgx_lwt.S and is shared across the different lwt backends